src/bucket/BucketManagerImpl.cpp

```diff
@@ -584,6 +612,20 @@ BucketManagerImpl::getReferencedBuckets() const
             }
         }
     }
+
+    return referenced;
+}
+
+std::set<Hash>
+BucketManagerImpl::getAllReferencedBuckets() const
+{
+    ZoneScoped;
+    auto referenced = getBucketListReferencedBuckets();
+    if (!mApp.getConfig().MODE_ENABLES_BUCKETLIST)
```

Comment:
I don't think I understand why this function was split in two / why this early return happens here.

Reply:
This has to do with freeing indexes, see comment on `freeIndex`.